src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -932,6 +932,77 @@ class CreateTableStatementSegment(BaseSegment):
     )
 
 
+@postgres_dialect.segment()
+class CreateTableAsStatementSegment(BaseSegment):
+    """A `CREATE TABLE AS` statement.
+
+    As specified in https://www.postgresql.org/docs/13/sql-createtableas.html
+    """
+
+    type = "create_table_as_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        OneOf(
+            Sequence(
+                OneOf("GLOBAL", "LOCAL", optional=True),
+                Ref("TemporaryGrammar", optional=True),
+            ),
+            "UNLOGGED",
+            optional=True,
+        ),
+        "TABLE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        AnyNumberOf(
+            Sequence(
+                Bracketed(
+                    Delimited(
+                        Sequence(
+                            Ref("ColumnReferenceSegment"),
+                            AnyNumberOf(Ref("ColumnConstraintSegment")),
+                        ),
+                    ),
+                ),
+                optional=True,
+            ),
+            Sequence("USING", Ref("FunctionSegment"), optional=True),
+            OneOf(
+                Sequence(
+                    "WITH",
+                    Bracketed(
+                        AnyNumberOf(
+                            Sequence(
+                                Ref("ParameterNameSegment"),
+                                Sequence(
+                                    Ref("EqualsSegment"),
+                                    Ref("LiteralGrammar"),
+                                ),
+                                optional=True,
+                            )
+                        )
+                    ),
+                    optional=True,
+                ),
+                Sequence("WITHOUT", "OIDS", optional=True),
+            ),
+            Sequence(
+                "ON",
+                "COMMIT",
+                OneOf("PRESERVE ROWS", "DELETE ROWS", "DROP"),
+                optional=True,
+            ),
+            Sequence("TABLESPACE", Ref("ParameterNameSegment"), optional=True),
+        ),
+        "AS",
+        OneOf(
+            OptionallyBracketed(Ref("SelectableGrammar"))
+            # TODO Add TABLE, VALUES, AND EXECUTE commands
```

Comment:
These might be as easy as this to add:

```suggestion
            Sequence("TABLE", Ref("TableReferenceSegment"),
            Ref("ValuesClauseSegment"),
            Sequence("EXECUTE", Ref("FunctionSegment")),
```

Though should add some test examples if willing to add this.

Reply:
I think this is worth adding in this PR, as it's not a complicated change

Reply:
This has been added.